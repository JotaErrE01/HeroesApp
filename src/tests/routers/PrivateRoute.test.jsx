import { prettyDOM, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import PrivateRoute from "../../routers/PrivateRoute";

describe('Pruebas en <PrivateRoute />', () => {

    const rest = {
        location: {
            pathname: '/dc'
        }
    }

    Storage.prototype.setItem = jest.fn();
    
    test('Debe de mostrar el componente si esta autenticado y guardar local storage', () => {
        const { container } = render(
            <MemoryRouter>
                <PrivateRoute 
                    isAuthenticated={ true }
                    component={() => <span>Listo</span>}
                    {...rest}
                />
            </MemoryRouter>
        );

        expect( container.firstChild ).toMatchSnapshot();
        expect( container.firstChild.textContent ).toBe('Listo');
        expect( localStorage.setItem ).toHaveBeenCalledWith('lastPath', rest.location.pathname);
    })

    test('Debe de bloquear el componente si no esta autenticado', () => {
        const { container, debug } = render(
            <MemoryRouter>
                <PrivateRoute 
                    isAuthenticated={ false }
                    component={() => <span>Listo</span>}
                    {...rest}
                />
            </MemoryRouter>
        );
        
        expect( container.firstChild ).toBeNull();
        expect( container.querySelector('span') ).not.toBeInTheDocument();
    })
    
    
})