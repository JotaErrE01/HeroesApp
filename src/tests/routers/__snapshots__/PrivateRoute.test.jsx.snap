// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrivateRoute /> Debe de mostrar el componente si esta autenticado y guardar local storage 1`] = `
<span>
  Listo
</span>
`;
